app/controllers/purchases_controller.rb

```diff
@@ -42,4 +37,14 @@ def limit
   def purchase_params
     params.require(:purchase).permit(:date, :description, :amount, :payee, :subcategory)
   end
+
+  def purchase_sort
```

Comment:
I think this method could be better named - `limit_purchases`, `filter_purchases`, maybe?